test/llvm_ir_correct/omp_atomic_reduce_ice.f90

```diff
@@ -0,0 +1,49 @@
+
+! RUN: %flang -fopenmp -Hy,69,0x1000 -S -emit-llvm %s -o - | FileCheck %s
+! RUN: %flang -i8 -fopenmp -Hy,69,0x1000 -S -emit-llvm %s -o - | FileCheck %s
```

Comment:
Isn't `XBIT(69,0x1000)` off by default? Does it need to be explicitly turned off for this test?

Reply:
@bryanpkc To enable the feature that flag has to be switched off. Please see the definition of use_opt_atomic below.

```
#define OPT_OMP_ATOMIC !XBIT(69,0x1000)
...
LOGICAL use_opt_atomic(int d)
{
#ifdef OMP_OFFLOAD_LLVM
  return is_in_omptarget(d);
#endif
  return OPT_OMP_ATOMIC;
}
```

https://github.com/flang-compiler/flang/blob/997feeef4b15e3dbea4be8b6309869ba64991548/tools/flang1/flang1exe/semsmp.c#L551
https://github.com/flang-compiler/flang/blob/997feeef4b15e3dbea4be8b6309869ba64991548/tools/flang1/flang1exe/semsmp.c#L10324

Reply:
Apologies, I misunderstood your question. 

The flag is enabled by default in the flang-driver. Hence it needs to be turned off for this test.
https://github.com/flang-compiler/flang-driver/blob/f4508c0261f0c0a5920d850e6d0a769fadd17e00/lib/Driver/ToolChains/Flang.cpp#L279

Reply:
Thanks for the clarification.

Reply:
I have one more question about the comment in Flang.cpp. What's preventing the option from being turned on by default? What does it take to provide a named option?

Reply:
1) It would required libatomic to be present.
2) There are a few issues including the one fixed in this ticket. See github issue below.
https://github.com/flang-compiler/flang/issues/677

Reply:
@bryanpkc does that sound OK? Can you approve?
Ping @shivaramaarao

Reply:
Thanks for the answer and sorry for the late reply.